integrationTests/src/test/java/org/sonarqube/gradle/AbstractGradleIT.java

```diff
@@ -168,13 +164,33 @@ private static int getJavaVersion() {
     return Integer.parseInt(version);
   }
 
+  private static Properties getDumpedProperties(List<String> command) throws IOException {
+    for (String part : command) {
+      if (part.trim().startsWith("-Dsonar.scanner.internal.dumpToFile=")) {
+        File dumpFile = new File(part.split("=")[1]);
+        return loadProperties(dumpFile);
+      }
+    }
+    return null;
+  }
+
+  private static Properties loadProperties(File out) throws IOException {
+    Properties props = new Properties();
+    try (FileReader fr = new FileReader(out)) {
+      props.load(fr);
+    }
+    return props;
+  }
+
   protected static class RunResult {
     private final String log;
     private final int exitValue;
+    private final Properties dumpedProperties;
```

Comment:
We may want to use an optional value to show explicitly that it may be missing

```suggestion
    private final Optional<Properties> dumpedProperties;
```

Reply:
Wille be addressed as part of another ticket